rune/libenclave/internal/runtime/pal/skeleton/sgxsign.c

```diff
@@ -280,16 +289,41 @@ static bool measure_encl(const char *path, uint8_t *mrenclave)
 			goto out;
 	}
 
+	uint64_t *ssa_frame =  malloc(4096 * ssa_frame_size);
+	if (ssa_frame == NULL){
+		fprintf(stderr, "Failed malloc memory for ssa_frame\n");
+		return false;
+	}
+	memset(ssa_frame, 0, 4096 * ssa_frame_size);
```

Comment:
Actually you really don't need to allocate a buffer for ssa frame, because data[] is 4096 and it can be used as a zero page to be used in your eadd for each ssa page.

Just do like this:
```
memset(data, 0, sizeof(data));

flags = SGX_SECINFO_REG | SGX_SECINFO_R | SGX_SECINFO_W | SGX_SECINFO_X;
for (offset = sb.st_size; offset < sb.st_size + PAGE_SIZE * ssa_frame_size; offset += PAGE_SIZE) {
    if (!mrenclave_eadd(ctx, offset, flags))
        goto out_saa;

    if (!mrenclave_eextend(ctx, offset, data))
        goto out_saa;
}
```

What a simple and clear implementation!